src/components/Map/Map/Map.js

```diff
@@ -23,11 +23,30 @@ export const Map = React.forwardRef((props, ref) => {
     const fitBounds = React.useCallback(() => {
         if (!mapRef.current || !mapRef.current.state.map) return;
         const { map } = mapRef.current.state;
+        const { LatLngBounds, Circle: CircleClass, Geocoder } = window.google.maps;
+        const geocoder = new Geocoder();
+        const bounds = new LatLngBounds();
 
-        const { LatLngBounds, Circle: CircleClass } = window.google.maps;
-
-        if (markers.length) {
-            const bounds = new LatLngBounds();
+        if (markers.length === 1 && markers[0].description) {
+            const marker = markers[0];
+            if (marker.radius) {
+                const circle = new CircleClass({
+                    center: marker.center,
+                    radius: marker.radius,
+                });
+                bounds.union(circle.getBounds());
+                map.fitBounds(bounds);
+            } else if (marker.description) {
+                geocoder.geocode({ address: marker.description }, (result, status) => {
+                    if (status === 'OK') {
+                        map.fitBounds(result[0].geometry.viewport);
+                    }
+                });
+            } else {
+                bounds.extend(marker.center);
+                map.fitBounds(bounds);
+            }
+        } else if (markers.length) {
```

Comment:
it is hard to grasp what is going on in this method `fitBounds`. We have here 12 different conditions, it is hard to read. I think it should be refactored.

Reply:
makes sense, I'll refactor it